editor/filesystem_dock.cpp

```diff
@@ -1275,6 +1275,7 @@ void FileSystemDock::_try_duplicate_item(const FileOrFolder &p_item, const Strin
 	// Ensure folder paths end with "/".
 	String old_path = (p_item.is_file || p_item.path.ends_with("/")) ? p_item.path : (p_item.path + "/");
 	String new_path = (p_item.is_file || p_new_path.ends_with("/")) ? p_new_path : (p_new_path + "/");
+	const_cast<FileSystemDock *>(this)->path = new_path;
```

Comment:
What if the new path is invalid? I see several failure scenarios below.

Reply:
Why not move this operation to the end of the method after the duplication has succeeded, instead of doing it preemptively?

Reply:
Some paths return on success.

Reply:
Well you can at least move it after the first block of early returns.
```
	if (new_path == old_path) {
		return;
	} else if (old_path == "res://") {
		EditorNode::get_singleton()->add_io_error(TTR("Cannot move/rename resources root."));
		return;
	} else if (!p_item.is_file && new_path.begins_with(old_path)) {
		// This check doesn't erroneously catch renaming to a longer name as folder paths always end with "/".
		EditorNode::get_singleton()->add_io_error(TTR("Cannot move a folder into itself.") + "\n" + old_path + "\n");
		return;
	}
```